aws-iot-custommetric/src/main/java/com/amazonaws/iot/custommetric/UpdateHandler.java

```diff
@@ -95,12 +97,23 @@ void updateTags(AmazonWebServicesClientProxy proxy,
         // Yet we should, otherwise the resource wouldn't equate the template.
         Set<Tag> currentTags = listTags(proxy, resourceArn, logger);
 
-        // getDesiredResourceTags includes model+stack-level tags, reference: https://tinyurl.com/y2p8medk
-        Set<Tag> desiredTags = Translator.translateTagsToSdk(request.getDesiredResourceTags());
-        // TODO: uncomment this after we update the service to allow these (only from CFN)
-        // SystemTags are the default stack-level tags with aws:cloudformation prefix
-        // desiredTags.addAll(Translator.translateTagsToSdk(request.getSystemTags()));
-
+        // Combine all tags in one map that we'll use for the request
+        Map<String, String> allDesiredTagsMap = new HashMap<>();
+        if (request.getDesiredResourceTags() != null) {
+            // DesiredResourceTags includes both model and stack-level tags.
+            // Reference: https://tinyurl.com/yyxtd7w6
+            allDesiredTagsMap.putAll(request.getDesiredResourceTags());
+        }
+        if (request.getSystemTags() != null) {
+            // There are also system tags provided separately.
+            // SystemTags are the default stack-level tags with aws:cloudformation prefix.
+            allDesiredTagsMap.putAll(request.getSystemTags());
+        } else {
+            // System tags should never get updated as they are the stack id, stack name,
+            // and logical resource id.
+            logger.log("Unexpectedly, system tags are null in the update request for " + resourceArn);
+        }
```

Comment:
Shouldn't we add this check to "CreateHandler" as well (every Create should have system tags as well)?

Reply:
Yes, done